lib/web_ui/lib/src/engine/keyboard_binding.dart

```diff
@@ -211,10 +228,28 @@ class FlutterHtmlKeyboardEvent {
 // [dispatchKeyData] as given in the constructor. Some key data might be
 // dispatched asynchronously.
 class KeyboardConverter {
-  KeyboardConverter(this.performDispatchKeyData, {this.onMacOs = false});
+  KeyboardConverter(this.performDispatchKeyData, OperatingSystem platform)
+    : onMacOs = platform == OperatingSystem.macOs,
+      _mapping = _mappingFromPlatform(platform);
 
   final DispatchKeyData performDispatchKeyData;
+  // Whether the current platform is macOS, which affects how certain key events
+  // are comprehended.
   final bool onMacOs;
+  // Maps logical keys from key event properties.
+  final locale_keymap.LocaleKeymap _mapping;
+
+  static locale_keymap.LocaleKeymap _mappingFromPlatform(OperatingSystem platform) {
+    switch (platform) {
+      case OperatingSystem.iOs:
+      case OperatingSystem.macOs:
+        return locale_keymap.LocaleKeymap.darwin();
+      case OperatingSystem.windows:
+        return locale_keymap.LocaleKeymap.win();
+      default:
```

Comment:
The `default` case should be avoided: It is better to have cases for each enum entry so that if a new value is added, the linter will highlight this location as one that doesn't handle the new value.

https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values